bin/prepare-release.sh

```diff
@@ -1,15 +1,35 @@
 set -e
 
+function error {
+  RED='\033[0;31m'
+  NONE='\033[0m'
+  printf "$RED$1$NONE\n"
+  exit 1
+}
+
 if [ -z "$1" ]; then
-    echo "Provide a new version, current version is $(jq '.version' package.json)"
-    exit 1
+    error "Provide a new version, current version is $(jq '.version' package.json)"
 fi
 
 VERSION=$1
+RELEASE_BRANCH="release-$VERSION"
+
+CURRENT_BRANCH=$(git rev-parse --symbolic-full-name --abbrev-ref HEAD)
+if [[ "$CURRENT_BRANCH" != "main" ]] && [[ "$CURRENT_BRANCH" != "$RELEASE_BRANCH" ]]; then
+  error "You must be on branch main"
+fi
 
-git checkout main
+# Make sure we're up-to-date with origin.
 git fetch
-git pull origin main
+git pull --no-rebase origin main
```

Comment:
Should we use `--ff-only` while pulling. If that fails, we should halt as that would indicate a problem?

Reply:
👍 Now using `--ff-only`. 

> If that fails, we should halt as that would indicate a problem?

AFAICT this is already the case thanks to the `set -e` at the start of the script.